Description: doc: add option to not build the book
 mdbook is not available in all distributions, and other documentation does
 not require it. Add a meson "book" option (default true) to gate the book
 build, falling back to a no-op when mdbook is not found.
Author: Yaakov Selkowitz <yselkowi@redhat.com>
Origin: upstream, https://gitlab.freedesktop.org/wayland/wayland/-/commit/6c0a03a8fade7386d202dde1b2ace76dd6897417
Bug: https://gitlab.freedesktop.org/wayland/wayland/-/commit/6c0a03a8fade7386d202dde1b2ace76dd6897417
Forwarded: not-needed
Last-Update: 2026-07-29

--- a/doc/meson.build
+++ b/doc/meson.build
@@ -6,7 +6,7 @@ dot = find_program('dot')
 doxygen = find_program('doxygen')
 xsltproc = find_program('xsltproc')
 xmlto = find_program('xmlto')
-mdbook = find_program('mdbook')
+mdbook = find_program('mdbook', required: get_option('book'))
 
 cmd = run_command(doxygen, '--version', check: true)
 message('doxygen: ' + cmd.stdout().strip())
@@ -40,4 +40,6 @@ publican_html_dir = 'html'
 
 subdir('doxygen')
 subdir('publican')
-subdir('book')
+if mdbook.found()
+	subdir('book')
+endif
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -22,6 +22,10 @@ option('dtd_validation',
   description: 'Validate the protocol DTD (requires libxml2)',
   type: 'boolean',
   value: true)
+option('book',
+  description: 'Build the Wayland Book (requires mdbook)',
+  type: 'boolean',
+  value: true)
 option('icon_directory',
   description: 'Location used to look for cursors (defaults to ${datadir}/icons if unset)',
   type: 'string',
